### PR TITLE
Custom Date Formats

### PR DESCRIPTION
**Pull request by icasdri**
**tl;dr** This changes the way date formatting is handled, adding a new preference that allows selecting between different pre-written date formats as well as providing one's own custom format string. This allows the user to decide how to display the dates & times in any number of ways per their individual tastes. Note: this requires an update to the translations.

See **Motivation** below if you're interested.
Note: the changes included here need Translation work, see **What Still Needs To Be Done** for details.

For licensing purposes: the code and text provided are Copyright (c) 2016 icasdri, and released as free software. You can redistribute and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.

## The Changes
1. Remove the two existing date formatting preferences: namely *Imperial Dates* and *Show Day of Week*
2. Add a new *Date Format* preference which provides a list of date formats (that cover all the possibilities of the old *Imperial Dates* and *Show Day of Week*) and also includes a *Custom* option.
3. Add a new *Custom Date Format* preference, which takes a [SimpleDateFormat](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html) string -- this string is used if the *Custom* option is selected above.
4. Update logic in `WeatherRecyclerAdapter` accordingly, as well as values in `strings.xml` and `arrays.xml`. An error message is displayed (in place of an actual formatted date) if the custom format string provided by the user is not valid.

These changes have been tested to work on a real device running Android 5.0.1

## What Still Needs To Be Done
Translations. I have only modified the English `strings.xml`, as such, other languages will not work.
The following newly-added strings (followed by their current English values) need to be translated: 

    <string name="error_dateFormat">DATE FORMAT ERROR</string>
    <string name="setting_dateFormat">Date Format</string>
    <string name="setting_dateFormatCustom">Custom Date Format</string>
    <string name="setting_dateFormatCustom_summary">If Custom selected above, provide the custom SimpleDateFormat string to be used</string>
Additionally, the old `setting_imperialDate`, `setting_imperialDate_description`, and `setting_showDay` need to be removed (this has already been done in the English `strings.xml`)

## Motivation
First of all, I would like to give my thanks to all the devs that have contributed so far for this open-source, simple, and intuitive weather app. 

In using the app, I encountered the wish to display times in AM/PM format -- I live in the U.S. and I'll admit, we tell time weird :). **Forecastie** already has two options in its Settings concerning the display of dates & time in the long term forecast: namely the *Imperial Dates* and *Show Day of Week* options -- however, obviously none of these concern my issue. Instead of cluttering the Settings with even more date formatting options, I am pull-requesting a change that allows for the use of any arbitrary format string, both replacing and extending the existing options. I have since found a rather short format string that I now use (for the curious, it is `E - hh:mm a` = "Wed. - 01:00 PM")

### CODE REVIEW

**Comment by martykan** [comment]
Wow, thanks. That's a nice and much needed fix. The translations usually take a bit longer, so I will release it with only some languages and wait for other contributors to translate it afterwards.
